src/buildercore/cfngen.py

```diff
@@ -711,7 +715,14 @@ def regenerate_stack(stackname, **more_context):
     download_cloudformation_template(stackname)
     (pname, instance_id) = core.parse_stackname(stackname)
     more_context['stackname'] = stackname # TODO: purge this crap
-    more_context['alt-config'] = instance_id
+    # lsh@2019-09-27: usage of `instance_id` here is wrong. `instance_id` looks like "foobar" in "journal--foobar"
+    # and is only correct when an alt-config matches. We typically have alt-configs for our common environments, like
+    # ci, end2end, prod, continuumtest and has thus worked stably for a while now.
+    # ad-hoc instances whose instance-id does not match an environment will have it's alt-config ignored.
+    # the alt-config used during instance creation is found in `current_context` (but may not have always been the case).
+    #more_context['alt-config'] = instance_id
+    #more_context['alt-config'] = current_context.get('alt-config', instance_id)
+    more_context['alt-config'] = current_context['alt-config']
```

Comment:
This is what should ideally be happening, but I don't know how backwards compatible it is. It's such a minute and easily forgettable thing that I think we should set it to the correct value here and not think about again until if/when it breaks.